lib/btc.js

```diff
@@ -0,0 +1,179 @@
+      let hashItem = await hashDb.findOneAsync(
+        { _id: hashIdNode },
+        { _id: 1, nodeUri: 1 }
+      )
+      if (!hashItem) {
+        throw new Error('hash_id_node not found')
+      }
+
+      let targetNodeUri = hashItem.nodeUri || null
+      if (!targetNodeUri) {
+        throw new Error('node_uri for hash_id_node not found')
+      }
+
+      // make sure proof hasnt expired
+      if (!utils.hashIdExpired(PROOF_EXPIRE_MINUTES, hashIdNode)) {
+        let proofHandles = [{ uri: targetNodeUri, hashIdNode: hashIdNode }]
+        await updateCmd.updateHashesByHashIdNodeAsync(hashDb, proofHandles)
+      }
+
+      // retrieve the updated proof by hash_id_node
+      hashItem = await hashDb.findOneAsync({ _id: hashIdNode })
+      if (!hashItem || !hashItem.proof) throw new Error(`proof data not found`)
+
+      let evaluateResults
+      try {
+        evaluateResults = await EvaluateProofs([hashItem.proof])
+      } catch (error) {
+        console.error(`ERROR : Could not evaluate proof`)
+        return
+      }
+
+      displayOutputResults(evaluateResults, output, quiet, json)
+    } catch (error) {
+      output.addErrorResult({
+        hash_id_node: hashIdNode,
+        message: error.message
+      })
+      output.display(quiet, json)
+    }
+  } else {
+    // open storage and process update using hashes from local storage
+    let hashDb
+    let proofHandleSegments = []
+    let hashItems
+    try {
+      hashDb = await storage.connectAsync()
+      proofHandleSegments = await updateCmd.buildProofHandleSegments(hashDb)
+
+      let updateTasks = []
+      for (let x = 0; x < proofHandleSegments.length; x++) {
+        updateTasks.push(async () => {
+          return updateCmd.updateHashesByHashIdNodeAsync(
+            hashDb,
+            proofHandleSegments[x]
+          )
+        })
+      }
+      if (updateTasks.length > 0) {
+        try {
+          await parallel(updateTasks, 20)
+        } catch (error) {
+          console.error(`ERROR : Could not update proof data`)
+        }
+      }
+    } catch (error) {
+      output.addErrorResult({
+        message: error.message
+      })
+      output.display(quiet, json)
+      return
+    }
+
+    // retrieve updated hash_id_nodes
+    hashItems = await hashDb.findAsync({}, { _id: 1, proof: 1 })
+
+    // filter out all hashes with no proof data
+    hashItems = hashItems.filter(hashItem => {
+      return hashItem.proof !== null
+    })
+
+    // get proof array
+    let proofs = hashItems
+      .map(hashItem => {
+        return hashItem.proof
+      })
+      .filter(proof => {
+        return proof
+      })
+
+    let evaluateResults
+    try {
+      evaluateResults = await EvaluateProofs(proofs)
+    } catch (error) {
+      console.error(`ERROR : Could not evaluate proofs`)
+      return
+    }
+
+    displayOutputResults(evaluateResults, output, quiet, json)
+  }
+}
+
+function displayOutputResults(resultsArray, outputObj, quiet, json) {
```

Comment:
Main difference from `evaluate` is what we're outputting of course. We're only outputting proofs that have btc anchors. Uses bcoin's `TX.fromRaw` to parse the raw transaction hex.